src/vma/dev/net_device_table_mgr.cpp

```diff
@@ -323,10 +344,122 @@ void net_device_table_mgr::verify_ipoib_mode(struct ifaddrs* ifa)
+				break;
+			}
+			qp_init_attr.recv_cq = cq;
+			qp_init_attr.send_cq = cq;
+			struct ibv_qp* qp = ibv_create_qp(p_ib_ctx->get_ibv_pd(), &qp_init_attr);
+			if (qp) {
+				success = true;
+				IF_VERBS_FAILURE(ibv_destroy_qp(qp)) {
+					ndtm_logdbg("qp destroy failed on interface %s (errno=%d %m)", ifname, errno);
+				} ENDIF_VERBS_FAILURE;
+				IF_VERBS_FAILURE(ibv_destroy_cq(cq)) {
+					ndtm_logdbg("cq destroy failed on interface %s (errno=%d %m)", ifname, errno);
+				} ENDIF_VERBS_FAILURE;
+				IF_VERBS_FAILURE(ibv_destroy_comp_channel(channel)) {
+					ndtm_logdbg("channel destroy failed on interface %s (errno=%d %m)", ifname, errno);
+				} ENDIF_VERBS_FAILURE;
+			} else {
+				ndtm_logdbg("QP creation failed on interface %s (errno=%d %m), Traffic will not be offloaded \n", ifname, errno);
+			}
+			break;
+		}
+	}
+	rdma_free_devices(pp_ibv_context_list);
+	return success;
+}
+
+bool net_device_table_mgr::check_raw_qp_privliges(char* ifname)
+{
+
+	int ret = validate_raw_qp_privliges();
+	if (ret == 0) {
+		//// MLNX_OFED raw_qp_privliges file exist with bad value
+		vlog_printf(VLOG_WARNING,"******************************************************************************\n");
+		vlog_printf(VLOG_WARNING,"* Verbs RAW_PACKET QP type creation is limited for root user access          *\n");
+		vlog_printf(VLOG_WARNING,"* Working in this mode might causes VMA malfunction over Ethernet interfaces *\n");
+		vlog_printf(VLOG_WARNING,"* Interface %s will not be offloaded.                                         \n", ifname);
+		vlog_printf(VLOG_WARNING,"* WARNING: the following steps will restart your network interface!          *\n");
+		vlog_printf(VLOG_WARNING,"* 1. \"echo options ib_uverbs disable_raw_qp_enforcement=1 > /etc/modprobe.d/ib_uverbs.conf\" *\n");
+		vlog_printf(VLOG_WARNING,"* 2. \"/etc/init.d/openibd restart\"                                           *\n");
+		vlog_printf(VLOG_WARNING,"* Read the RAW_PACKET QP root access enforcement section in the VMA's User Manual for more information *\n");
+		vlog_printf(VLOG_WARNING,"******************************************************************************\n");
+		return false;
+	} else if (ret == -1) {
+		// raw_qp_privliges from MLNX_OFED file doesn't exist so check upstream for CAP_NET_RAW
+		if (!verify_user_has_CAP_NET_RAW_privliges()) {
+			vlog_printf(VLOG_WARNING,"******************************************************************************\n");
+			vlog_printf(VLOG_WARNING,"* Usage is restricted to root or user with CAP_NET_RAW privileges.           *\n");
+			vlog_printf(VLOG_WARNING,"* Interface %s will not be offloaded.                                         \n", ifname);
+			vlog_printf(VLOG_WARNING,"* Read the CAP_NET_RAW and root access section in the VMA's User Manual for more information *\n");
+			vlog_printf(VLOG_WARNING,"******************************************************************************\n");
+			return false;
```

Comment:
I don't think we should add it here since this code only checks for permissions and has nothing to do with qp creation. it's true that the flow that will get us here started with qp creation failure but this function is more general than that